src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -477,14 +477,43 @@
     FrameClauseUnitGrammar=OneOf("ROWS", "RANGE"),
     # It's as a sequence to allow to parametrize that in Postgres dialect with LATERAL
     JoinKeywords=Sequence("JOIN"),
-    TableConstraintReferenceOptionGrammar=OneOf(
+    ReferentialActionGrammar=OneOf(
         "RESTRICT",
         "CASCADE",
         Sequence("SET", "NULL"),
         Sequence("NO", "ACTION"),
         Sequence("SET", "DEFAULT"),
     ),
     DropBehaviorGrammar=OneOf("RESTRICT", "CASCADE", optional=True),
+    ReferenceDefinitionGrammar=Sequence(
```

Comment:
Think it'd be worth making this a segment instead? It could make future rules easier to write.

Reply:
Hmm, I can't see these specific segments being referenced in a rule? `CASE`/`WHEN`/`THEN`/`ELSE` is pretty common, whereas this is a bit niche. I guess we could move things like this to a separate segment but feels like it might be overkill (at least for now).

Happy for a follow up PR to make them into segments if you want though

Reply:
No worries, just thinking aloud. And yes, almost all the rules are focused on queries or DML.